src/Operators/remapping.jl

```diff
@@ -108,6 +110,35 @@ function overlap_weights(
     return kron(Y_ov, X_ov)
 end
 
+function overlap_weights(
+    target::T,
+    source::S,
+) where {
+    T <: SpectralElementSpace1D{<:IntervalTopology, Quadratures.GL{1}},
+    S <: SpectralElementSpace1D{<:IntervalTopology, Quadratures.GL{1}},
+}
+    target_topo = Spaces.topology(target)
+    source_topo = Spaces.topology(source)
+
+    nx1 = Topologies.nlocalelems(target_topo)
+    nx2 = Topologies.nlocalelems(source_topo)
+
+    X_ov = spzeros(nx1, nx2)
+    
+    for i in 1:nx1
+        vertices_i = Topologies.vertex_coordinates(target_topo, i)
+        xmin_i, xmax_i = vertices_i[1].x, vertices_i[2].x
+        for j in 1:nx2
+            vertices_j = Topologies.vertex_coordinates(source_topo, j)
+            xmin_j, xmax_j = vertices_j[1].x, vertices_j[2].x
+            xmin_ov, xmax_ov = max(xmin_i, xmin_j), min(xmax_i, xmax_j)
+            x_overlap = xmax_ov > xmin_ov ? xmax_ov - xmin_ov : continue
+            X_ov[i, j] = x_overlap
+        end
+    end
```

Comment:
Some code duplication with 2D case. Could abstract and create function for this (the Y-direction case for 2D is the annoying part) if we feel this is worth it, but this all may be temporary with FV <-> Spectral coming.

Reply:
You're right. This code block that finds the overlap in x-dimension could be wrapped in a function and used both in the 1D and 2D cases. I don't think it will be unnecessary, even if we move to FV <-> SEM remapping. I think we could still keep the FV <-> FV prototype, no?

Reply:
Let me know what you think of the change here @valeriabarra!

Reply:
I like it! Much cleaner and better organized. Thanks!!